internal/status/status_by_name.go

```diff
@@ -0,0 +1,38 @@
+package status
+
+import (
+	"sync"
+)
+
+//UserStatus stores the map of users to be disabled.
+type UserStatus struct {
+	sync.RWMutex
+	DisalbeUserMap map[string]struct{}
+}
+
+// NewUserStatus creates an instance of Status
+func NewUserStatus() (Service, error) {
+	return &UserStatus{DisalbeUserMap: make(map[string]struct{})}, nil
+}
+
+// CheckUserStatus check the status of user
+func (t *UserStatus) CheckUserStatus(user string) (bool, error) {
```

Comment:
will this function ever return an error other than nil
can we use just
```golang
func (t *UserStatus) CheckUserStatus(user string) bool
```
the error will make sense if we are going to have
a complex implementaiton of **type Service interface** 	CheckUserStatus(user string) (bool, error) in the future